connector_node/risingwave-source-test/src/test/java/com/risingwave/connector/MySQLSourceTest.java

```diff
@@ -0,0 +1,192 @@
+            fail("SQL exception: ", e);
+        }
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        connectorServer.shutdown();
+        mysql.stop();
+    }
+
+    @Test
+    public void testLines() throws InterruptedException, SQLException {
+        Lock lock = new ReentrantLock();
+        Condition done = lock.newCondition();
+        Connection connection = SourceTestClient.connect(mysql);
+        String query = testClient.sqlStmts.getProperty("tpch.create.orders");
+        SourceTestClient.performQuery(connection, query);
+        query =
+                "LOAD DATA INFILE '/home/orders.tbl' "
+                        + "INTO TABLE orders "
+                        + "CHARACTER SET UTF8 "
+                        + "FIELDS TERMINATED BY '|' LINES TERMINATED BY '\n';";
+        SourceTestClient.performQuery(connection, query);
+        Iterator<GetEventStreamResponse> eventStream =
+                testClient.getEventStreamStart(mysql, SourceType.MYSQL, "test", "orders");
+        AtomicInteger count = new AtomicInteger();
+        Thread t1 =
+                new Thread(
+                        () -> {
+                            while (eventStream.hasNext()) {
+                                List<CdcMessage> messages = eventStream.next().getEventsList();
+                                for (CdcMessage ignored : messages) {
+                                    count.getAndIncrement();
+                                }
+                                if (count.get() == 10000) {
+                                    lock.lock();
+                                    try {
+                                        done.signal();
+                                    } finally {
+                                        lock.unlock();
+                                    }
+                                }
+                            }
+                        });
+        t1.start();
+        lock.lock();
+        try {
+            done.await();
+        } finally {
+            lock.unlock();
+        }
+        logger.info("count: {}", count.get());
+        assertThat(count.get()).isEqualTo(10000);
+        connection.close();
+    }
+
+    @Ignore
+    @Test
+    public void getTestJson() throws InterruptedException, SQLException {
+        Connection connection = SourceTestClient.connect(mysql);
+        String query =
+                "CREATE TABLE IF NOT EXISTS orders ("
+                        + "O_KEY BIGINT NOT NULL, "
+                        + "O_BOOL BOOLEAN, "
+                        + "O_TINY TINYINT, "
+                        + "O_INT INT, "
+                        + "O_REAL REAL, "
+                        + "O_DOUBLE DOUBLE, "
+                        + "O_DECIMAL DECIMAL(15, 2), "
+                        + "O_CHAR CHAR(15), "
+                        + "O_DATE DATE, "
+                        + "O_TIME TIME, "
+                        + "O_TIMESTAMP TIMESTAMP, "
```

Comment:
please cover corner cases
for date: `1000-01-01` and `9999-12-31`
for time: `-838:59:59` and `838:59:59`
for datetime: `1000-01-01 00:00:00` and `9999-12-31 23:59:59`
for timestamp: `'1970-01-01 00:00:01' UTC` and `'2038-01-09 03:14:07' UTC` (and other timezones)

Reply:
thanks for your comments bohan! i am wondering what is the expected behavior when converting mysql time to postgres time (since risingwave is postgres compatible)...

https://support.huaweicloud.com/intl/en-us/productdesc-drs/drs_01_0324.html

> TIME values supported by MySQL range from '-838:59:59' to '838:59:59'. For details, see the [official MySQL documentation](https://dev.mysql.com/doc/refman/5.7/en/time.html). For PostgreSQL, the minimum value of the TIME type is 00:00:00 and the maximum value is 24:00:00. In MySQL, if a value of the TIME type is less than 00:00:00 or greater than 24:00:00, DRS will convert it to 00:00:00.

I guess we can either let it overflow, or convert it to 00:00:00 as above...?

Reply:
If the value is out of range of RisingWave's type, which means the data may have been corrupted in upstream. Thus we can report an out-of-range (e.g. underflow, overflow) error to log and skip this row.

Reply:
Sure, I mean we should cover them in test cases and document the behavior.